PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsHostRunConfigurationProducer.kt

```diff
@@ -0,0 +1,85 @@
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.jetbrains.plugins.azure.functions.run
+
+import com.intellij.execution.actions.ConfigurationContext
+import com.intellij.execution.actions.LazyRunConfigurationProducer
+import com.intellij.execution.configurations.ConfigurationTypeUtil
+import com.intellij.openapi.util.Ref
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.psi.PsiElement
+import com.jetbrains.rider.model.RunnableProjectKind
+import com.jetbrains.rider.model.runnableProjectsModel
+import com.jetbrains.rider.projectView.solution
+import com.jetbrains.rider.run.configurations.getSelectedProject
+
+class AzureFunctionsHostRunConfigurationProducer
+    : LazyRunConfigurationProducer<AzureFunctionsHostConfiguration>() {
+
+    override fun getConfigurationFactory() =
+            ConfigurationTypeUtil.findConfigurationType(AzureFunctionsHostConfigurationType::class.java)
+                    .configurationFactories
+                    .single()
+
+    override fun isConfigurationFromContext(
+            configuration: AzureFunctionsHostConfiguration,
+            context: ConfigurationContext
+    ) : Boolean {
+        val selectedProject = context.getSelectedProject() ?: return false
+        val projects = context.project.solution.runnableProjectsModel.projects.valueOrNull ?: return false
+
+        val selectedProjectFilePath = FileUtil.toSystemIndependentName(selectedProject.getFile()?.path ?: "")
+        val runnableProject = projects.firstOrNull {
+            it.kind == RunnableProjectKind.AzureFunctions &&
+            it.projectFilePath == selectedProjectFilePath &&
+            configuration.parameters.projectFilePath == selectedProjectFilePath
+        }
+
+        return runnableProject != null
+    }
+
+    override fun setupConfigurationFromContext(
+            configuration: AzureFunctionsHostConfiguration,
+            context: ConfigurationContext,
+            ref: Ref<PsiElement>
+    ): Boolean {
+        val selectedProject = context.getSelectedProject() ?: return false
+        val projects = context.project.solution.runnableProjectsModel.projects.valueOrNull ?: return false
+
+        val selectedProjectFilePath = FileUtil.toSystemIndependentName(selectedProject.getFile()?.path ?: "")
```

Comment:
`selectedProject.getFile()?.path ?: ""`
You set `selectedProjectFilePath` to invalid value if project path is not set. I guess you can fail earlier and make all these checks earlier and skip redundant calls to `FileUtil.toSystemIndependentName` and further:

```
val selectedProjectPath = context.getSelectedProject()?.getFile()?.path  ?: return false
```